src/server/frontend_wayland/output_manager.cpp

```diff
@@ -25,86 +25,33 @@
 
 namespace mf = mir::frontend;
 namespace mg = mir::graphics;
+namespace mw = mir::wayland;
 using namespace mir::geometry;
 
-mf::Output::Output(wl_display* display, mg::DisplayConfigurationOutput const& initial_configuration) :
-    output{make_output(display)},
-    current_config{initial_configuration}
-{
-}
-
-mf::Output::~Output()
-{
-    // Notify all clients that the wl_output has gone away
-    wl_global_destroy(output);
-    /*
-     * The above call doesn't release the wl_resources any client
-     * has bound, merely tells them that the global has gone away.
-     * The server-side wl_resources have to remain valid so that
-     * the client can call wl_output::release on it.
-     *
-     * We therefore need to ensure that destroying the wl_resource-s
-     * doesn't result in attempting to access fields of the now-destroyed
-     * Output.
-     */
-    for (auto const& client : resource_map)
-    {
-        for (auto* resource : client.second)
-        {
-            wl_resource_set_destructor(resource, [](auto) {});
-        }
-    }
-}
-
-void mf::Output::handle_configuration_changed(mg::DisplayConfigurationOutput const& config)
-{
-    for (auto const& client : resource_map)
-    {
-        for (auto const& resource : client.second)
-        {
-            // Possibly not optimal
-            send_initial_config(resource, config);
-        }
-    }
-}
-
-void mf::Output::for_each_output_resource_bound_by(wl_client* client, std::function<void(wl_resource*)> const& functor)
-{
-    auto const resources = resource_map.find(client);
-
-    if (resources != resource_map.end())
-    {
-        for (auto const& resource : resources->second)
-        {
-            functor(resource);
-        }
-    }
-}
-
 namespace
 {
-auto as_subpixel_arrangement(MirSubpixelArrangement arrangement) -> wl_output_subpixel
+auto as_subpixel_arrangement(MirSubpixelArrangement arrangement) -> uint32_t
```

Comment:
It does not. I believe making bitwise operations work was complex, and not deemed worth it at the time since no real bugs have come out of the lack of type safety yet..